java/maven.hints/src/org/netbeans/modules/maven/hints/errors/EnablePreviewMavenProj.java

```diff
@@ -52,144 +41,97 @@
+
+        ActionConfig[] actions = new ActionConfig[] {
+            ActionConfig.runAction("run"), // NOI18N
+            ActionConfig.runAction("debug"), // NOI18N
+            ActionConfig.runAction("profile"), // NOI18N
+            ActionConfig.runAction("run.single.main"), // NOI18N
+            ActionConfig.runAction("debug.single.main"), // NOI18N
+            ActionConfig.runAction("profile.single.main"), // NOI18N
+            ActionConfig.testAction("test"), // NOI18N
+            ActionConfig.testAction("test.single"), // NOI18N
+            ActionConfig.testAction("debug.test.single"), // NOI18N
+            ActionConfig.testAction("profile.test.single"), // NOI18N
+        };
+        for (ActionConfig action : actions) {
+            NetbeansActionMapping mapp = ModelHandle2.getMapping(action.actionName, prj, cfg);
+            Map<String, String> properties = mapp.getProperties();
+            String existingValue = properties.getOrDefault(action.propertyName, "");
+
+            if (!existingValue.contains(ENABLE_PREVIEW_FLAG)) {
+                properties.put(action.propertyName, ENABLE_PREVIEW_FLAG + (existingValue .isEmpty() ? "" : " ") + existingValue);
+                ModelHandle2.putMapping(mapp, prj, cfg);
             }
 
-            return null;
         }
     }
 
-    private boolean isMavenProject(Project prj) {
-        if (prj == null) {
-            return false;
+    private static final class ActionConfig {
+        public final String actionName;
+        public final String propertyName;
+
+        public ActionConfig(String actionName, String propertyName) {
+            this.actionName = actionName;
+            this.propertyName = propertyName;
         }
-        FileObject prjDir = prj.getProjectDirectory();
-        if (prjDir == null) {
-            return false;
+        public static ActionConfig runAction(String actionName) {
+            return new ActionConfig(actionName, "exec.args");
+        }
+        public static ActionConfig testAction(String actionName) {
+            return new ActionConfig(actionName, "argLine");
         }
-
-        FileObject pom = prjDir.getFileObject("pom.xml");
-        return (pom != null) && pom.isValid();
-
     }
 
     private static class AddMvnCompilerPluginForEnablePreview implements ModelOperation<POMModel> {
 
         private static final String MAVEN_COMPILER_GROUP_ID = "org.apache.maven.plugins"; // NOI18N
         private static final String MAVEN_COMPILER_ARTIFACT_ID = "maven-compiler-plugin"; // NOI18N
         private static final String COMPILER_ID_PROPERTY = "compilerId"; // NOI18N
+        private static final String SOURCE = "source"; // NOI18N
+        private static final String TARGET = "target"; // NOI18N
         private static final String COMPILER_ARG = "compilerArgs"; // NOI18N
         private static final String MAVEN_COMPILER_VERSION = "3.3"; // NOI18N
```

Comment:
can you bump this to `3.11.0`?

`3.11.0` still supports running on JDK 8 but it knows about the release options too. So the user will see the `release` option hint right away in the pom.